### PR TITLE
Add additional tests for CUDA and L0 providers

### PR DESCRIPTION
**Pull request by vinser52**
### Description
To improve code coverage the new tests for CUDA and L0 providers are added. In next PR this tests can be refactored to avoid code duplications.

### Checklist
- [x] Code compiles without errors locally
- [x] All tests pass locally
- [x] CI workflows execute properly

### CODE REVIEW

File: test/providers/provider_level_zero.cpp [hunk 1]
```diff
@@ -222,6 +210,97 @@ TEST_P(umfLevelZeroProviderTest, basic) {
...
+
+    umf_result = umfMemoryProviderGetMinPageSize(provider, nullptr, nullptr);
+    ASSERT_EQ(umf_result, UMF_RESULT_ERROR_INVALID_ARGUMENT);
+
+    umf_result = umfMemoryProviderGetRecommendedPageSize(provider, 0, nullptr);
+    ASSERT_EQ(umf_result, UMF_RESULT_ERROR_INVALID_ARGUMENT);
+
+    umfMemoryProviderDestroy(provider);
+}
+
```
Comment (bratpiorka) [inline test/providers/provider_level_zero.cpp:303]: please remove TODO from L305 :)

File: test/providers/provider_level_zero.cpp [hunk 2]
```diff
@@ -167,21 +162,14 @@ struct umfLevelZeroProviderTest
         }
 
         ASSERT_NE(zeMemoryTypeExpected, ZE_MEMORY_TYPE_UNKNOWN);
-
-        memAccessor = accessor;
     }
 
-    void TearDown() override {
-        int ret = destroy_context(hContext);
```
Comment (bratpiorka) [inline test/providers/provider_level_zero.cpp:175]: where do you call destroy context?

**Review by ldorau — CHANGES_REQUESTED** [review]
Could you add tests for `cu_memory_provider_get_last_native_error` either?
`cu_memory_provider_get_last_native_error` and `cu_store_last_native_error` are not used at all.

**Comment by ldorau** [comment]
> Could you add tests for `cu_memory_provider_get_last_native_error` either? `cu_memory_provider_get_last_native_error` and `cu_store_last_native_error` are not used at all.

It can be in a separate (next) PR if you prefer.

**Review by ldorau — APPROVED** [review]
> Could you add tests for `cu_memory_provider_get_last_native_error` either? `cu_memory_provider_get_last_native_error` and `cu_store_last_native_error` are not used at all.

It can be in a separate (next) PR if you prefer.

File: test/providers/provider_level_zero.cpp (hunk 1 above)
Comment (vinser52) [reply test/providers/provider_level_zero.cpp:303]: Done, removed.

Right now only the following `TODO` comments remains:
```c
// TODO add tests that mixes Level Zero Memory Provider and Disjoint Pool
```

File: test/providers/provider_level_zero.cpp (hunk 2 above)
Comment (vinser52) [reply test/providers/provider_level_zero.cpp:175]: `l0Params_device_memory`, `l0Params_shared_memory`, `l0Params_host_memory` are global variables, see L306 - L311. So it is an error to destroy the context in the `TearDown` method because it is called after each test.

The same is true for the CUDA tests

**Comment by vinser52** [comment]
> > Could you add tests for `cu_memory_provider_get_last_native_error` either? `cu_memory_provider_get_last_native_error` and `cu_store_last_native_error` are not used at all.
> 
> It can be in a separate (next) PR if you prefer.

Yeah, better to add them in a separate PR. Need to think about the scenario when CUDA provider returns provider-specific error.

**Review by lukaszstolarczuk — APPROVED** [review]
